CleverPush/Source/CleverPushInstance.m

```diff
@@ -469,8 +472,10 @@ - (void)applicationWillEnterForeground API_AVAILABLE(ios(10.0)) {
                 [CPLog debug:@"CleverPushInstance: applicationWillEnterForeground: There is no subscription for CleverPush SDK."];
         } else {
             if (!notificationsEnabled && !ignoreDisabledNotificationPermission) {
-                [CPLog info:@"notification authorization revoked, unsubscribing"];
-                [self unsubscribe];
+                if (autoRequestNotificationPermission) {
```

Comment:
Not required here for unsubscribe

Reply:
Same thing as above.

If the user sets the `autoRequestNotificationPermission` to false, they will minimize the application, and if the user relaunches the application, this function `applicationWillEnterForeground` will be called, and it will go into this block because the user has not given access to push notification permission, so it will return false for this `notificationsEnabled` method, and the `unsubscribe` function will be called.

![Screenshot 2023-11-21 at 4 46 54 PM](https://github.com/cleverpush/cleverpush-ios-sdk/assets/110810122/15bff9e2-47fc-4a34-8104-ab0178070150)